tirocks-sys/build.rs

```diff
@@ -0,0 +1,261 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use cc::Build;
+use cmake::Config;
+use std::path::{Path, PathBuf};
+use std::{env, str};
+
+// On these platforms jemalloc-sys will use a prefixed jemalloc which cannot be linked together
+// with RocksDB.
+// See https://github.com/gnzlbg/jemallocator/blob/bfc89192971e026e6423d9ee5aaa02bc56585c58/jemalloc-sys/build.rs#L45
+const NO_JEMALLOC_TARGETS: &[&str] = &["android", "dragonfly", "musl", "darwin"];
+
+// Generate the bindings to rocksdb C-API.
+// Try to disable the generation of platform-related bindings.
+#[cfg(feature = "update-bindings")]
+fn bindgen_rocksdb(file_path: &Path) {
+    println!("cargo:rerun-if-env-changed=TEST_BIND");
+    let gen_tests = env::var("TEST_BIND").map_or(false, |s| s == "1");
+
+    let mut builder = bindgen::Builder::default();
+
+    if env::var("CARGO_CFG_TARGET_OS").map_or(false, |s| s == "windows") {
+        builder = builder.clang_arg("-D _WIN32_WINNT=0x600");
```

Comment:
what's it for. Please add a comment.

Reply:
It's copied from grpc-rs. Some file may require to define minimal supported windows version at compile time.